olxcleaner/loader/xml.py

```diff
@@ -109,14 +109,21 @@ def read_course(edxobj, node, directory, filename, errorstore, htmlfiles, pointe
                     empty = False
 
     # Check for a pointer tag
+    node_filename = os.path.splitext(os.path.split(filename)[-1])[0]
     if empty and edxobj.is_pointer(node.attrib):
-
         if pointer:
-            # The target of a pointer tag cannot be a pointer itself
-            errorstore.add_error(SelfPointer(filename,
-                                             edxobj=edxobj))
-            edxobj.broken = True
-            return
+            if node.attrib.get('url_name') == node_filename:
+                # The target of a pointer tag cannot be a pointer itself
+                errorstore.add_error(SelfPointer(filename,
```

Comment:
more readable in one liner.
`errorstore.add_error(SelfPointer(filename, edxobj=edxobj))`

Reply:
There is a duplicate code we can clean with something as follows, or whatever makes more sense?
```
node_pointing_to_itself = node.attrib.get('url_name') == node_filename

if node_pointing_to_itself:
   pointer_error = SelfPointer
else:
  pointer_error = PointerAlreadyPointedAt

errorstore.add_error(pointer_error(filename, edxobj=edxobj))
edxobj.broken = True
return
```